sources/models/src/modeled_types/oci_defaults.rs

```diff
@@ -120,10 +157,155 @@ mod oci_defaults_rlimits {
     fn linux_rlimit_strings() {
-        check_rlimit_strings(
-            OciDefaultsResourceLimitType::MaxOpenFiles,
-            "max-open-files",
-            "RLIMIT_NOFILE",
-        );
+        let rlimits = [
+            (MaxAddressSpace, "max-address-space", "RLIMIT_AS"),
+            (MaxCoreFileSize, "max-core-file-size", "RLIMIT_CORE"),
+            (MaxCpuTime, "max-cpu-time", "RLIMIT_CPU"),
+            (MaxDataSize, "max-data-size", "RLIMIT_DATA"),
+            (MaxFileLocks, "max-file-locks", "RLIMIT_LOCKS"),
+            (MaxFileSize, "max-file-size", "RLIMIT_FSIZE"),
+            (MaxLockedMemory, "max-locked-memory", "RLIMIT_MEMLOCK"),
+            (MaxMsgqueueSize, "max-msgqueue-size", "RLIMIT_MSGQUEUE"),
+            (MaxNicePriority, "max-nice-priority", "RLIMIT_NICE"),
+            (MaxOpenFiles, "max-open-files", "RLIMIT_NOFILE"),
+            (
+                MaxPendingSignals,
+                "max-pending-signals",
+                "RLIMIT_SIGPENDING",
+            ),
+            (MaxProcesses, "max-processes", "RLIMIT_NPROC"),
+            (
+                MaxRealtimePriority,
+                "max-realtime-priority",
+                "RLIMIT_RTPRIO",
+            ),
+            (MaxRealtimeTimeout, "max-realtime-timeout", "RLIMIT_RTTIME"),
+            (MaxResidentSet, "max-resident-set", "RLIMIT_RSS"),
+            (MaxStackSize, "max-stack-size", "RLIMIT_STACK"),
+        ];
+
+        for rlimit in rlimits {
+            check_rlimit_strings(rlimit);
+        }
+    }
+}
+
+/// OciDefaultsResourceLimitValue specifies what all values are
+/// allowed to be set for rlimit in default spec.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub enum OciDefaultsResourceLimitValue {
+    Value(u32),
+    Unlimited,
+}
+
+impl OciDefaultsResourceLimitValue {
+    pub fn to_u64(self) -> u64 {
+        match self {
+            OciDefaultsResourceLimitValue::Value(x) => x as u64,
+            OciDefaultsResourceLimitValue::Unlimited => u64::MAX,
+        }
+    }
+}
+
+impl Serialize for OciDefaultsResourceLimitValue {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match *self {
+            OciDefaultsResourceLimitValue::Value(x) => serializer.serialize_u32(x),
+            OciDefaultsResourceLimitValue::Unlimited => serializer.serialize_str("unlimited"),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for OciDefaultsResourceLimitValue {
+    fn deserialize<D>(
+        deserializer: D,
+    ) -> std::result::Result<OciDefaultsResourceLimitValue, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let json_value = Value::deserialize(deserializer)?;
+
+        return match &json_value {
+            Value::Number(number) => {
+                if number.is_u64() {
+                    let s = number.to_string().parse::<u32>().map_err(|e| {
+                        de::Error::custom(format!(
```

Comment:
I believe we use `Snafu` for these errors in `modeled_types` so I would like to use that for these errors.

Reply:
Serde expect errors while deserializing  in form of D::Error, hence we need to return the error in form of D::Error.

Reply:
These custom errors only in Serde deserializers makes sense.